src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1688,53 +1674,50 @@ internal static class Defaults
         /// <param name="initLearningRate">The initial learning rate used by SGD.</param>
         /// <param name="l2Weight">The L2 regularizer constant.</param>
         /// <param name="loss">The loss function to use.</param>
-        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
-        public StochasticGradientDescentClassificationTrainer(IHostEnvironment env,
+        internal StochasticGradientDescentClassificationTrainer(IHostEnvironment env,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
             string weightColumn = null,
```

Comment:
Can this be set to Defaults.WeightColumn?

Reply:
we need to take closer look at how 'weights' are being used, or if they are being used at all

please see #2175.

---
In reply to: [248867308](https://github.com/dotnet/machinelearning/pull/2093#discussion_r248867308) [](ancestors = 248867308)